src/engine/qcommon/gameinfo.h

```diff
@@ -0,0 +1,63 @@
+/*
+===========================================================================
+
+Daemon GPL Source Code
+Copyright (C) 2017, Daemon Developers
+
+This file is part of the Daemon GPL Source Code (Daemon Source Code).
+
+Daemon Source Code is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Daemon Source Code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Daemon Source Code.  If not, see <http://www.gnu.org/licenses/>.
+
+===========================================================================
+*/
+
+#ifndef GAMEINFO_H_
+#define GAMEINFO_H_
+
+#include <iostream>
+
+
+class Gameinfo
```

Comment:
I know it would probably need more work, but is it possible to leverage existing cfg file code?
also some of the variables already exists as cvar we can set as command line option, maybe we can make the engine reading those cvars first and, if command line option is set, override.

Reply:
I've looked into the existing cfg file parsing code again. I don't think we can leverage this code as it simply treats every line as a console command. The gameinfo.cfg doesn't contain commands and I don't think that NAME, VERSION, DEFAULT_BASEPAK or GAMENAME_STRING should be cvars. Adding commands just for those is way to complicated for what I try to achieve.
I renamed that file to gameinfo.conf to make it obvious that it is not a cfg file that contains commands and cvars.

About the cvars: MASTERSERVER<num> and WWW_BASEURL could probably dropped completely from this config file since they are only default values.

Reply:
OK, then it looks OK to me as reusing the cfg parsing code looks to be the over-engineered way.
Also, by not using cfg parsing we have less thing to think about security: we are sure non-cvar variable cannot be abused with cvar code.